alib/run_experiment.py

```diff
@@ -363,26 +368,60 @@ def _spawn_process(self, args):
             # else:
             #     log.info("Process with index {} is currently still active".format(process_id))
 
+    def _handle_finished_process(self, scenario_id, execution_id, process_index, failed):
+        self.finished_tasks.append((scenario_id, execution_id, failed))
+        # joining without timeout might cause the whole execution to wait for the slowest process until the errors/results of other
+        # processes are handled or new ones created.
+        self.processes[process_index].join()
+        # terminate might corrupt the queues or interrupt exception handling...
+        # self.processes[process_index].terminate()
+        self.processes[process_index] = None
+        self.currently_active_processes -= 1
+        # set the current scenario to invalid, it is used when processing the result, which should be done before calling this function
+        self.current_scenario[process_index] = None
+
     def _retrieve_results_and_spawn_processes(self):
         while self.currently_active_processes > 0:
-            result = self.result_queue.get()
-            scenario_id, execution_id, alg_result, process_index = result
-
-            self._process_result(result)
-            self.finished_tasks.append((scenario_id, execution_id))
+            try:
```

Comment:
thanks for adding error handling; normally, we simply investigated the errors in the log files of the spawned workers!